src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AdminController.cs

```diff
@@ -128,8 +127,11 @@ public class AdminController : Controller
         {
             if (!await _authorizationService.AuthorizeAsync(User, Permissions.ManageOpenIdApplications))
                 return Unauthorized();
-            
-            if (model.UpdateClientSecret && string.IsNullOrWhiteSpace(model.ClientSecret))
+
+            if (model.Type == ClientType.Public && !string.IsNullOrEmpty(model.ClientSecret))
+                ModelState.AddModelError(nameof(model.ClientSecret), "No client secret can be set for public applications.");
+
```

Comment:
I always use braces for if/else if/else states, but I followed the same code style for consistency here.